### PR TITLE
refactor: extract RunPlan as its own domain entity

### PR DESCRIPTION
**Pull request by ptab**
Split from https://github.com/Vandebron/mpyl/pull/400 to make reviewing easier.

### CODE REVIEW

File: src/mpyl/cli/repository.py [hunk 1]
```diff
@@ -160,10 +162,12 @@ def init(obj: RepoContext, url: str, pull: int, branch: str, pristine: bool):
 
     console.log("Preparing repository for a new run...")
 
-    with Repository.from_shallow_diff_clone(
-        branch.replace("refs/heads/", ""), url, "main", obj.config, Path(".")
-    ) if pristine else Repository(
-        config=RepoConfig.from_config(parse_config(obj.config))
+    with (
```
Comment (Jorg88) [inline src/mpyl/cli/repository.py:165]: I've seen these changes more often in your pr's, just wondering: is this Black or your IDE settings?

File: src/mpyl/run_plan.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+from .project import Stage
+from .project_execution import ProjectExecution
+
+
+@dataclass(frozen=True)
+class RunPlan:
+    plan: dict[Stage, set[ProjectExecution]]
+
+    @staticmethod
+    def empty() -> "RunPlan":
```
Comment (Jorg88) [inline src/mpyl/run_plan.py:14]: I have not seen this `"RunPlan"` syntax before, what does it do?

**Review by Jorg88 — APPROVED** [review]

File: src/mpyl/cli/repository.py (hunk 1 above)
Comment (ptab) [reply src/mpyl/cli/repository.py:165]: I'm not sure. I have IntelliJ configured to run Black on save:
image
With (what I think are) the default settings:
image

Not sure why it ends up with different styling 🤷

File: src/mpyl/run_plan.py (hunk 2 above)
Comment (ptab) [reply src/mpyl/run_plan.py:14]: Python is not very good at dealing with forward references (i.e. using types that haven't been defined yet), so in order to deal with this we need to quote the type hint to make it a literal and tell the interpreter to resolve it later. You can find this explained in more detail in https://peps.python.org/pep-0484/#forward-references and https://peps.python.org/pep-0484/#the-problem-of-forward-declarations.

**Pull request merged**